### PR TITLE
Ownership Cache and Extensions

### PR DESCRIPTION
**Pull request by detaybey**
**What this PR does / why we need it**:
Added Ownership memory cache and an extension to query it. Main purpose of this cache is to return the recent Owner for the requested Resource path. As it heavily depends on several querying of the Ownership table, we used a memory cache.

### CODE REVIEW

File: src/backend/joseki.be/webapp/Database/Cache/OwnershipCache.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        /// Returns the Ownership entries.
+        /// In this scenario it is updated daily, so the expiration is set to 1 day.
+        /// If the ownership will be updated manually, this cache should be invalidated.
+        /// </summary>
+        public async Task<List<OwnershipEntity>> GetEntries()
+        {
+            if (!this.cache.TryGetValue<IEnumerable<OwnershipEntity>>(CacheName, out var entries))
+            {
+                var db_entries = await this.db.Set<OwnershipEntity>().AsNoTracking().ToListAsync();
+                entries = this.cache.Set(CacheName, db_entries, absoluteExpiration: DateTimeOffset.Now.AddDays(1));
```
Comment (v1r7u) [inline src/backend/joseki.be/webapp/Database/Cache/OwnershipCache.cs:46]: Great, that you used expiration time 👍 
Could you only check if the method expects UTC time?

Comment (v1r7u) [inline src/backend/joseki.be/webapp/Database/Cache/OwnershipCache.cs:46]: What about storing in cache only <component_id, owner> tuple? Likely, It decreases used memory in a few times.

File: src/backend/joseki.be/webapp/Handlers/OwnershipExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    public enum ComponentType
+    {
+        Kubernetes,
+        AzureSubscription,
+    }
+
+    /// <summary>
+    /// Base class for parsing ComponentId.
+    /// </summary>
+    public class ComponentId
```
Comment (v1r7u) [inline src/backend/joseki.be/webapp/Handlers/OwnershipExtensions.cs:66]: Nice to see this as a separate class finally. I had to do it this way from the very beginning...
What about moving it to a dedicated file?

File: src/backend/joseki.be/webapp/Database/Cache/OwnershipCache.cs (hunk 1 above)
Comment (detaybey) [reply src/backend/joseki.be/webapp/Database/Cache/OwnershipCache.cs]: docs [don't tell much](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-3.1) about the timezone of expiration. I decided to use `absoluteExpirationRelativeToNow: TimeSpan.FromDays(1)`

Comment (detaybey) [reply src/backend/joseki.be/webapp/Database/Cache/OwnershipCache.cs]: Done, using _ValueTuples_ now.

File: src/backend/joseki.be/webapp/Handlers/OwnershipExtensions.cs (hunk 2 above)
Comment (detaybey) [reply src/backend/joseki.be/webapp/Handlers/OwnershipExtensions.cs]: Moved under /Database/Models/ComponentId.cs, looks good?

**Pull request merged**